src/renderer/viz/expressions/viewportFeatures.js

```diff
@@ -0,0 +1,90 @@
+import BaseExpression from './base';
+import * as schema from '../../../renderer/schema';
+import Property from './basic/property';
+
+/**
+ * Generates a list of features in the viewport
+ *
+ * For each feature, the properties specified as arguments to this expression will be available.
+ * Filtered features will not be present in the list.
+ * This expression cannot be used in the visualization, it can only be used in JavaScript code as in the example below.
+ *
+ * @param {...Property} properties - properties that will appear in the feature list
+ * @return {ViewportFeatures} ViewportFeatures
+ *
+ * @example <caption>Define and use a list of features. (String)</caption>
+ * const source = carto.source.Dataset('data');
+ * const viz = new carto.Viz(`
+ *          \@list: viewportFeatures($value, $category)
+ * `);
+ * const layer = carto.Layer('layer', source, viz);
+ * ...
+ *
+ * layer.on('updated', () => {
+ *     viz.variables.list.value.forEach(feature => {
+ *         console.log('value:', feature.value, 'category:', feature.category);
+ *     });
+ * });
+ *
+ * @memberof carto.expressions
+ * @name viewportFeatures
+ * @function
+ * @api
+ */
+export class ViewportFeatures extends BaseExpression {
+    constructor(...properties) {
+        if (properties.some(p => !(p instanceof Property))) {
```

Comment:
Right now it is not supported (we throw an error in the viewportFeatures if all parameters are not strictly properties).
I'm working on that case right now but I'm encountering some difficulties: if the variable is not used in the viz other than in the viewportFeatures it is not resolved (`_resolveAliases`not called). So I prefer to merge this as it is (so it can be tested) and deal with that in another ticket.

Reply:
`_resolveAliasesnot ` is not being called because you didn't put the children in the call to the `super` constructor and the method is not overridden either :laughing: